public/components/trace_analytics/home.tsx

```diff
@@ -141,20 +150,40 @@ export const Home = (props: HomeProps) => {
   };
 
   const dataSourceMenuComponent = useMemo(() => {
-    return (
-      <DataSourceMenu
-        setMenuMountPoint={props.setActionMenu}
-        componentType={'DataSourceSelectable'}
-        componentConfig={{
-          savedObjects: props.savedObjectsMDSClient.client,
-          notifications: props.notifications,
-          fullWidth: true,
-          onSelectedDataSources: onSelectedDataSource,
-          dataSourceFilter: dataSourceFilterFn,
-        }}
-      />
-    );
-  }, [props.setActionMenu, props.savedObjectsMDSClient.client, props.notifications]);
+    if (!dataSourceMenuSelectable) {
+      return (
+        <DataSourceMenuView
+          setMenuMountPoint={props.setActionMenu}
+          componentType={'DataSourceView'}
+          componentConfig={{
+            activeOption: dataSourceMDSId,
+            fullWidth: true,
+            dataSourceFilter: dataSourceFilterFn,
+          }}
+        />
+      );
+    } else {
+      return (
+        <DataSourceMenu
+          setMenuMountPoint={props.setActionMenu}
+          componentType={'DataSourceSelectable'}
+          componentConfig={{
+            savedObjects: props.savedObjectsMDSClient.client,
+            notifications: props.notifications,
+            fullWidth: true,
+            onSelectedDataSources: onSelectedDataSource,
+            dataSourceFilter: dataSourceFilterFn,
+            activeOption: dataSourceMDSId,
+          }}
+        />
+      );
+    }
+  }, [
```

Comment:
Maybe we do something like this:

```suggestion
const sharedProps = {
  setMenuMountPoint: props.setActionMenu,
  componentConfig: {
    activeOption: dataSourceMDSId,
    fullWidth: true,
    dataSourceFilter: dataSourceFilterFn,
  },
};

return dataSourceMenuSelectable ? (
  <DataSourceMenu
    {...sharedProps}
    componentType={'DataSourceSelectable'}
    componentConfig={{
      ...sharedProps.componentConfig,
      savedObjects: props.savedObjectsMDSClient.client,
      notifications: props.notifications,
      onSelectedDataSources: onSelectedDataSource,
    }}
  />
) : (
  <DataSourceMenuView
    {...sharedProps}
    componentType={'DataSourceView'}
  />
);
```

Reply:
Looks cleaner, thanks applied change!